core/node/eth_sender/src/eth_tx_manager.rs

```diff
@@ -334,16 +334,22 @@ impl EthTxManager {
             // that `tx` is not mined and we should resend it.
             // We only resend the first un-mined transaction.
             if operator_nonce.latest <= tx.nonce {
+                let last_sent_at_block = storage
+                    .eth_sender_dal()
+                    .get_block_number_on_last_sent_attempt(tx.id)
+                    .await
+                    .unwrap();
```

Comment:
is it safe to unwrap?

I'd really stick to the previous approach this one looks more error prone

Reply:
It's safe to unwrap as if a transaction is in-flight, then it has at least one send attempt. And maybe I'm missing something, but there is an unwrap on almost the same method call, but calculating first, not last attempt just 10 lines below.